### PR TITLE
Ug touch up

### PR DESCRIPTION
**Pull request by Gabriel4357**

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -461,8 +465,13 @@ _You can update your account details to keep your account secure by calling this
 **Note:**
 
 * At least one of the optional fields must be provided.
-* The `PASSWORD`{.swift} must be at least 8 characters long.
 * The details will be updated without checking against the current details.
+* `USERNAME`{.swift} must be alphanumeric.
+* `PASSWORD`{.swift} must be at least 8 characters long.
```
Comment (jianyangg) [inline docs/UserGuide.md]: Should `PASSWORD` be 8 alphanumeric characters long?

File: docs/UserGuide.md [hunk 2]
```diff
@@ -507,7 +516,9 @@ secret question that was set during account registration._
 
 * If `ANSWER`{.swift} is provided, `NEW_PASSWORD`{.swift} and `CONFIRM_PASSWORD`{.swift} must also be provided and vice
   versa.
-* `NEW_PASSWORD`{.swift} and `CONFIRM_PASSWORD`{.swift} must be at least 8 characters long.
+* `ANSWER`{.swift} can take any characters.
+* `NEW_PASSWORD`{.swift} must be at least 8 characters long.
```
Comment (jianyangg) [inline docs/UserGuide.md]: Alphanumeric?

File: docs/UserGuide.md [hunk 3]
```diff
@@ -561,13 +573,12 @@ _You can add a new customer's details into HomeBoss with this command._
...
 **Note:**
 
-* All fields are required.
 * HomeBoss does not allow you to add customers with the same phone number.
 * A unique customer ID (may not be consecutive) will be assigned to the customer.
-* The `NAME`{.swift} must be alphanumeric and can contain spaces.
-* The `PHONE_NUMBER`{.swift} must be exactly 8 digits.
-* The `EMAIL`{.swift} must follow the `local@domain`{.swift} format.
-* The `ADDRESS`{.swift} can take any values.
+* `NAME`{.swift} must be alphanumeric and can contain spaces.
```
Comment (jianyangg) [inline docs/UserGuide.md:560]: Should use just letters

File: docs/UserGuide.md [hunk 4]
```diff
@@ -586,23 +597,22 @@ _You can add a new customer's details into HomeBoss with this command._
...
 
 **Format:** `customer view CUSTOMER_ID`{.swift}
 
 <box type="note" background-color="#dff0d8" border-color="#d6e9c6" icon=":information_source:">
 
 **Note:**
 
-* All fields are required.
-* The `DELIVERY_ID`{.swfit} must be an integer.
+* `CUSTOMER_ID`{.swfit} must be an integer greater than 0.
```
Comment (jianyangg) [inline docs/UserGuide.md]: swift

File: docs/UserGuide.md [hunk 5]
```diff
@@ -638,8 +648,15 @@ you can search for the customer using keywords by calling this command._
 
 **Note:**
 
-* You must provide at least one keyword to search for the customer.
-* You can choose to provide multiple keywords.
+* You must provide at least one non-empty alphanumeric `KEYWORD`{.swift} to search for the Customer.
```
Comment (jianyangg) [inline docs/UserGuide.md]: You must provide at least one non-empty `KEYWORD` ...

File: docs/UserGuide.md [hunk 6]
```diff
@@ -662,12 +679,24 @@ worries! You can edit the details of a customer simply by calling this command.
...
 **Format:** `customer edit CUSTOMER_ID [--name NAME] [--phone PHONE_NUMBER] [--email EMAIL] [--address ADDRESS]`
 
-<box background-color="#dff0d8" border-color="#d6e9c6" type="info" icon=":information_source:">
-    **Note:** [ ] around a parameter indicates that it is optional.
+<box type="note" background-color="#dff0d8" border-color="#d6e9c6" icon=":information_source:">
+
+**Note:**
+
+* At least one of the optional fields must be provided.
+* `CUSTOMER_ID`{.swfit} must be an integer greater than 0.
```
Comment (juliusgambe) [inline docs/UserGuide.md]: `CUSTOMER_ID`{.swift} must be an integer greater than 0 that corresponds to an existing Customer in the address book.`

File: docs/UserGuide.md [hunk 7]
```diff
@@ -691,6 +720,15 @@ _You can delete any customers who no longer require your services from HomeBoss
 
 **Format:** `customer delete CUSTOMER_ID`{.swift}
 
+<box type="note" background-color="#dff0d8" border-color="#d6e9c6" icon=":information_source:">
+
+**Note:**
+
+* `CUSTOMER_ID`{.swfit} must be an integer greater than 0.
```
Comment (juliusgambe) [inline docs/UserGuide.md]: `CUSTOMER_ID`{.swift} must be an integer greater than 0 that corresponds to an existing Customer in the address book.

File: docs/UserGuide.md [hunk 8]
```diff
@@ -402,9 +402,12 @@ Format:** `register --user USERNAME --password PASSWORD --confirmPass CONFIRM_PA
...
 **Note:**
 
-* All fields are required.
-* The `PASSWORD`{.swift} must be at least 8 characters long.
-* The `SECRET_QUESTION`{.swift} and `ANSWER`{.swift} will be used for account recovery.
+* `USERNAME`{.swift} must be alphanumeric.
+* `PASSWORD`{.swift} must be at least 8 alphanumeric characters long.
+* `CONFIRM_PASSWORD`{.swift} must be the same as `PASSWORD`{.swift}.
+* `SECRET_QUESTION`{.swift} can take any characters.
+* `ANSWER`{.swift} can take any characters.
```
Comment (B-enguin) [inline docs/UserGuide.md]: Non-empty?

File: docs/UserGuide.md [hunk 9]
```diff
@@ -433,8 +436,9 @@ Note that you need to have an account registered with HomeBoss._
 
 **Note:**
 
-* All fields are required.
-  </box>
+* `USERNAME`{.swift} must be alphanumeric.
```
Comment (B-enguin) [inline docs/UserGuide.md]: Username can have space?

File: docs/UserGuide.md [hunk 10]
```diff
@@ -638,8 +648,15 @@ you can search for the customer using keywords by calling this command._
...
-* You must provide at least one keyword to search for the customer.
-* You can choose to provide multiple keywords.
+* You must provide at least one non-empty alphanumeric `KEYWORD`{.swift} to search for the Customer.
+* You can optionally provide additional keywords, a Customer name that matches any of the given keywords will be 
+  displayed.
+  For example, the keywords `Alex Tan`{.swift} will display `Alex Wong`{.swift}, `Alex Tan`{.swift},
+  and `Tan Ah Meng`{.swift}.
+* The keyword must exactly match any word in the Customer name. For example the keyword `Alex`{.swift} will match
+  `Alex`{.swift} but not `Alexander`{.swift}.
+* The search for keyword(s) given are case-insensitive
```
Comment (jianyangg) [inline docs/UserGuide.md]: Should this be a statement with a full-stop? Also, could we say "The search is not case sensitive."

File: docs/UserGuide.md (hunk 6 above)
Comment (jianyangg) [inline docs/UserGuide.md]: swift

File: docs/UserGuide.md [hunk 11]
```diff
@@ -864,6 +912,19 @@ delivery? No worries! Just edit the details of a delivery simply by using this f
 **Format:** `delivery edit DELIVERY_ID [--name DELIVERY_NAME] [--customer CUSTOMER_ID] [--date EXPECTED_DELIVERY_DATE]
 [--status STATUS] [--note NOTE]`
 
+<box type="note" background-color="#dff0d8" border-color="#d6e9c6" icon=":information_source:">
+
+**Note:**
+
+* At least one of the optional fields must be provided.
+* `DELIVERY_ID`{.swfit} must be an integer greater than 0.
```
Comment (juliusgambe) [inline docs/UserGuide.md]: `DELIVERY_ID`{.swift} must be an integer greater than 0 that corresponds to an existing Delivery in the address book.

File: docs/UserGuide.md [hunk 12]
```diff
@@ -662,12 +679,24 @@ worries! You can edit the details of a customer simply by calling this command.
...
+<box type="note" background-color="#dff0d8" border-color="#d6e9c6" icon=":information_source:">
+
+**Note:**
+
+* At least one of the optional fields must be provided.
+* `CUSTOMER_ID`{.swfit} must be an integer greater than 0.
+* `NAME`{.swift} must be alphanumeric and can contain spaces.
+* `PHONE_NUMBER`{.swift} must be exactly 8 digits.
+* `EMAIL`{.swift} must follow the `local@domain`{.swift} format.
+* `ADDRESS`{.swift} can take any characters.
```
Comment (B-enguin) [inline docs/UserGuide.md:667]: non-empty?

File: docs/UserGuide.md (hunk 7 above)
Comment (jianyangg) [inline docs/UserGuide.md]: swift

File: docs/UserGuide.md [hunk 13]
```diff
@@ -893,11 +954,22 @@ delivery? No worries! Just edit the details of a delivery simply by using this f
...
 
-**Format:** `delivery status CUSTOMER_ID STATUS`
+**Format:** `delivery status DELIVERY_ID STATUS`
+
+
+<box type="note" background-color="#dff0d8" border-color="#d6e9c6" icon=":information_source:">
+
+**Note:**
+
+* `DELIVERY_ID`{.swfit} must be an integer greater than 0.
```
Comment (juliusgambe) [inline docs/UserGuide.md]: `DELIVERY_ID`{.swift} must be an integer greater than 0 that corresponds to an existing Delivery in the address book.

File: docs/UserGuide.md [hunk 14]
```diff
@@ -922,8 +994,8 @@ note to a specific delivery.
 **Note:**
 
 * If the delivery already has a Note it will be overwritten by the new Note given.
-* The `DELIVERY_ID`{.swift} must be an Integer.
-* The `NOTE`{.swift} must a Non-empty Alphanumeric String.
+* `DELIVERY_ID`{.swift} must be an integer greater than 0.
```
Comment (juliusgambe) [inline docs/UserGuide.md]: `DELIVERY_ID`{.swift} must be an integer greater than 0 that corresponds to an existing Delivery in the address book.

File: docs/UserGuide.md [hunk 15]
```diff
@@ -755,9 +804,8 @@ view a more in-depth description of the delivery.
 
 **Note:**
 
-* All fields are required.
-* The `DELIVERY_ID`{.swift} must be an integer.
-  </box>
+* `DELIVERY_ID`{.swift} must be an integer greater than 0.
+</box>
```
Comment (B-enguin) [inline docs/UserGuide.md]: Corresponds to the valid delivery?

File: docs/UserGuide.md [hunk 16]
```diff
@@ -950,6 +1022,13 @@ been cancelled or completed. If that's the case, simply use this feature to dele
 
 **Format:** `delivery delete DELIVERY_ID`
 
+<box type="note" background-color="#dff0d8" border-color="#d6e9c6" icon=":information_source:">
+
+**Note:**
+
+* `DELIVERY_ID`{.swift} must be an integer greater than 0.
```
Comment (juliusgambe) [inline docs/UserGuide.md]: `DELIVERY_ID`{.swift} must be an integer greater than 0 that corresponds to an existing Delivery in the address book.

File: docs/UserGuide.md [hunk 17]
```diff
@@ -727,7 +765,18 @@ Cool huh! We try to help you speed up the process as much as possible.
...
-**Format:** `delivery add DELIVERY_NAME --customer CUSTOMER_ID --date DELIVERY_DATE`
+**Format:** `delivery add DELIVERY_NAME --customer CUSTOMER_ID --date EXPECTED_DELIVERY_DATE`
+
+<box type="note" background-color="#dff0d8" border-color="#d6e9c6" icon=":information_source:">
+
+**Note:**
+
+* `DELIVERY_NAME`{.swift} must be alphanumeric and can contain spaces.
+* `CUSTOMER_ID`{.swift} must be an integer greater than 0 that corresponds to an existing Customer in the address book.
+* `EXPECTED_DELIVERY_DATE`{.swift} must be in yyyy-MM-dd format and be after today's date OR `TODAY`{.swift} for
```
Comment (jianyangg) [inline docs/UserGuide.md]: Nitpicking but would it be better to mention today then after today's date

File: docs/UserGuide.md [hunk 18]
```diff
@@ -789,12 +837,12 @@ list of deliveries will be sorted by expected delivery date in descending order
...
 
-* All fields are optional.
-* The delivery `STATUS`{.swift} accepts the following values: `CREATED`{.swift}/`SHIPPED`{.swift}/`COMPLETED`{.swift}/
+* `STATUS`{.swift} accepts the following values: `CREATED`{.swift}/`SHIPPED`{.swift}/`COMPLETED`{.swift}/
   `CANCELLED`{.swift}.
-* The `CUSTOMER_ID`{.swift} must be an integer.
-* The `EXPECTED_DELIVERY_DATE`{.swift} must be in yyyy-MM-dd format or `TODAY`{.swift} for today’s date.
-* The `SORT`{.swift} accepts the following values: `ASC`{.swift}/`DESC`{.swift}.
+* `CUSTOMER_ID`{.swift} must be an integer.
+* `EXPECTED_DELIVERY_DATE`{.swift} must be in yyyy-MM-dd format and be after today's date OR `TODAY`{.swift} for
```
Comment (jianyangg) [inline docs/UserGuide.md]: Same as above

File: docs/UserGuide.md [hunk 19]
```diff
@@ -789,12 +837,12 @@ list of deliveries will be sorted by expected delivery date in descending order
...
 **Note:**
 
-* All fields are optional.
-* The delivery `STATUS`{.swift} accepts the following values: `CREATED`{.swift}/`SHIPPED`{.swift}/`COMPLETED`{.swift}/
+* `STATUS`{.swift} accepts the following values: `CREATED`{.swift}/`SHIPPED`{.swift}/`COMPLETED`{.swift}/
   `CANCELLED`{.swift}.
-* The `CUSTOMER_ID`{.swift} must be an integer.
-* The `EXPECTED_DELIVERY_DATE`{.swift} must be in yyyy-MM-dd format or `TODAY`{.swift} for today’s date.
-* The `SORT`{.swift} accepts the following values: `ASC`{.swift}/`DESC`{.swift}.
+* `CUSTOMER_ID`{.swift} must be an integer.
```
Comment (B-enguin) [inline docs/UserGuide.md]: Integer greater than 0 that corresponds to a valid customer?

File: docs/UserGuide.md [hunk 20]
```diff
@@ -864,6 +912,19 @@ delivery? No worries! Just edit the details of a delivery simply by using this f
 **Format:** `delivery edit DELIVERY_ID [--name DELIVERY_NAME] [--customer CUSTOMER_ID] [--date EXPECTED_DELIVERY_DATE]
 [--status STATUS] [--note NOTE]`
```
Comment (jianyangg) [inline docs/UserGuide.md]: Should we have the {.swift} styling here?

File: docs/UserGuide.md (hunk 11 above)
Comment (jianyangg) [inline docs/UserGuide.md]: swift

File: docs/UserGuide.md [hunk 21]
```diff
@@ -893,11 +954,22 @@ delivery? No worries! Just edit the details of a delivery simply by using this f
...
-**Format:** `delivery status CUSTOMER_ID STATUS`
+**Format:** `delivery status DELIVERY_ID STATUS`
+
+
+<box type="note" background-color="#dff0d8" border-color="#d6e9c6" icon=":information_source:">
+
+**Note:**
+
+* `DELIVERY_ID`{.swfit} must be an integer greater than 0.
+* `STATUS`{.swift} accepts the following values: `CREATED`{.swift}/`SHIPPED`{.swift}/`COMPLETED`{.swift}/
```
Comment (B-enguin) [inline docs/UserGuide.md:920]: CANCELLED?

File: docs/UserGuide.md [hunk 22]
```diff
@@ -922,8 +994,8 @@ note to a specific delivery.
 **Note:**
 
 * If the delivery already has a Note it will be overwritten by the new Note given.
-* The `DELIVERY_ID`{.swift} must be an Integer.
-* The `NOTE`{.swift} must a Non-empty Alphanumeric String.
+* `DELIVERY_ID`{.swift} must be an integer greater than 0.
+* `NOTE`{.swift} can take any characters.
```
Comment (B-enguin) [inline docs/UserGuide.md]: Note is alphanumeric

File: docs/UserGuide.md [hunk 23]
```diff
@@ -864,6 +912,19 @@ delivery? No worries! Just edit the details of a delivery simply by using this f
...
 
+<box type="note" background-color="#dff0d8" border-color="#d6e9c6" icon=":information_source:">
+
+**Note:**
+
+* At least one of the optional fields must be provided.
+* `DELIVERY_ID`{.swfit} must be an integer greater than 0.
+* `DELIVERY_NAME`{.swift} must be alphanumeric and can contain spaces.
+* `CUSTOMER_ID`{.swift} must be an integer greater than 0 that corresponds to an existing Customer in the address book.
+* `EXPECTED_DELIVERY_DATE`{.swift} must be in yyyy-MM-dd format and be after today's date OR `TODAY`{.swift} for
```
Comment (jianyangg) [inline docs/UserGuide.md]: Same as above

**Review by B-enguin — COMMENTED** [review]
Some comments

**Review by juliusgambe — COMMENTED** [review]
Changes suggested

File: docs/UserGuide.md [hunk 24]
```diff
@@ -875,11 +936,11 @@ delivery? No worries! Just edit the details of a delivery simply by using this f
 **Examples:**
 
 - `delivery edit 1 --status CANCELLED --note Sudden overseas business trip to attend to.` </br>
-  Edits Delivery 1's Delivery Status to CANCELLED and edits the delivery's note. (If delivery previously did not
-  have a note, a note will be created to the delivery)
+  Edits Delivery Status of the delivery ,with Delivery ID of 1, to CANCELLED and edits the note of the Delivery.
```
Comment (jianyangg) [inline docs/UserGuide.md]: Is the spacing around the comma weird? 
Edits Delivery Status of the **D**elivery, with Delivery ID of 1, to CANCELLED...

File: docs/UserGuide.md [hunk 25]
```diff
@@ -875,11 +936,11 @@ delivery? No worries! Just edit the details of a delivery simply by using this f
 **Examples:**
 
 - `delivery edit 1 --status CANCELLED --note Sudden overseas business trip to attend to.` </br>
```
Comment (jianyangg) [inline docs/UserGuide.md]: {.swift}

File: docs/UserGuide.md [hunk 26]
```diff
@@ -875,11 +936,11 @@ delivery? No worries! Just edit the details of a delivery simply by using this f
...
-  Edits Delivery 1's Delivery Status to CANCELLED and edits the delivery's note. (If delivery previously did not
-  have a note, a note will be created to the delivery)
+  Edits Delivery Status of the delivery ,with Delivery ID of 1, to CANCELLED and edits the note of the Delivery.
+  (If delivery does not have a note, a note will be created to the delivery)
 - `delivery edit 2 --name Vanilla Cake --customer 3` </br>
-  Edits Delivery 2's name to Vanilla Cake as it was misspelled and edits the customer to Customer 3 as User
-  accidentally typed in 2 instead. \
+  Edits Delivery with the ID of 2's Delivery Name to Vanilla Cake as it was misspelled and edits the customer to
```
Comment (jianyangg) [inline docs/UserGuide.md:895]: Edits the Delivery Name of the Delivery with ID of 2 ...

File: docs/UserGuide.md [hunk 27]
```diff
@@ -893,11 +954,22 @@ delivery? No worries! Just edit the details of a delivery simply by using this f
 I see that you've just shipped out an order. Nice! Now you can update the status of that particular delivery
 to SHIPPED! Using this feature, you can update the status of a delivery accordingly.
 
+
 <box background-color="#d9edf7" border-color="#bce8f1" type="tip" header="Tip">
     Delivery Status typed in need not be in uppercase.
 </box>
 
-**Format:** `delivery status CUSTOMER_ID STATUS`
+**Format:** `delivery status DELIVERY_ID STATUS`
```
Comment (jianyangg) [inline docs/UserGuide.md]: Are we missing {.swift}

File: docs/UserGuide.md (hunk 13 above)
Comment (jianyangg) [inline docs/UserGuide.md]: {.swift}

File: docs/UserGuide.md (hunk 21 above)
Comment (jianyangg) [inline docs/UserGuide.md:920]: CANCELLED status?

**Review by jianyangg — COMMENTED** [review]
Few nits to fix! Good efforts throughout.

**Review by juliusgambe — APPROVED** [review]
LGTM

**Pull request merged**